lib/panoptes/client/authentication.rb

```diff
@@ -0,0 +1,73 @@
+require 'deprecate'
+
+module Panoptes
+  class Client
+    module Authentication
+      extend Gem::Deprecate
+      def jwt_payload
+        raise NotLoggedIn unless @auth[:token]
+        payload, = JWT.decode @auth[:token], jwt_signing_public_key, algorithm: 'RS512'
+        payload
+      rescue JWT::ExpiredSignature
+        raise AuthenticationExpired
+      end
+
+      def token_contents
+        jwt_payload.fetch('data')
+      end
+
+      def token_expiry
+        jwt_payload.fetch('exp')
+      end
+
+      def authenticated?
+        token_contents['id'].present?
+      end
+
+      def authenticated_user_login
+        raise NotLoggedIn unless authenticated?
+        token_contents.fetch('login')
+      end
+
+      def authenticated_user_display_name
+        raise NotLoggedIn unless authenticated?
+        token_contents.fetch('dname')
+      end
+
+      def authenticated_user_id
+        raise NotLoggedIn unless authenticated?
+        token_contents.fetch('id')
+      end
+
+      def authenticated_admin?
+        raise NotLoggedIn unless authenticated?
+        token_contents.fetch('admin', false)
+      end
+
+
+      # @deprecated Please use {#token_contents} instead
+      def current_user
+        raise NotLoggedIn unless @auth[:token]
+
+        payload, = JWT.decode @auth[:token], jwt_signing_public_key, algorithm: 'RS512'
+        payload.fetch('data')
+      end
+      deprecate :current_user, :token_contents, 2019, 6
+
+      def jwt_signing_public_key
+        @jwt_signing_public_key ||= OpenSSL::PKey::RSA.new(File.read(@public_key_path))
+      end
+
+      def public_key_for_env(env)
```

Comment:
Is this used in this module or somewhere else? If elsewhere should it belong there?

Reply:
the hope is that we should only ever access the JWT from here, and that's the only thing that the client uses the public key for afaik